pkg/resources/grant_application_role_acceptance_test.go

```diff
@@ -1,44 +1,58 @@
 package resources_test
 
 import (
-	"context"
 	"fmt"
-	"strings"
 	"testing"
 
 	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
-	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
 
 	"github.com/hashicorp/terraform-plugin-testing/config"
-	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
-	"github.com/hashicorp/terraform-plugin-testing/terraform"
 	"github.com/hashicorp/terraform-plugin-testing/tfversion"
 )
 
+func createApp(t *testing.T, name string) *sdk.Application {
+	t.Helper()
+
+	stage, cleanupStage := acc.TestClient().Stage.CreateStage(t)
+	t.Cleanup(cleanupStage)
+
+	acc.TestClient().Stage.PutOnStage(t, stage.ID(), "TestAcc_GrantApplicationRole/manifest.yml")
+	acc.TestClient().Stage.PutOnStage(t, stage.ID(), "TestAcc_GrantApplicationRole/setup.sql")
+
+	applicationPackage, cleanupApplicationPackage := acc.TestClient().ApplicationPackage.CreateApplicationPackage(t)
+	t.Cleanup(cleanupApplicationPackage)
+
+	acc.TestClient().ApplicationPackage.AddApplicationPackageVersion(t, applicationPackage.ID(), stage.ID(), "v1")
+
+	application, cleanupApplication := acc.TestClient().Application.CreateApplicationWithID(t, sdk.NewAccountObjectIdentifier(name), applicationPackage.ID(), "v1")
```

Comment:
why do we need a certain name here? We want to steer away from that and leave the name/id generation to helpers whenever needed. It's possible to set it up that way here (application is returned so its id can be used to set up variables for the tf config file)

Reply:
Got it, fixed.